changelogs/fragments/5907-fix-gitlab_runner-not-idempotent.yml

```
@@ -0,0 +1,2 @@
+bugfixes:
+ - gitlab_runner - add missing parameter 'access_level' on runner creation and change default value to 'not_protected' (https://github.com/ansible-collections/community.general/issues/5907).
```

I would make this a bit more explicit what this means for users:
```suggestion
 - gitlab_runner - add missing parameter ``access_level`` on runner creation. It has been ignored on creation so far and was only used on updating (https://github.com/ansible-collections/community.general/issues/5907).
```
(I cut out the default change since that shouldn't be part of this PR anyway.)

---

Ok. sorry. I have not seen your comment. I have removed the default value of the parameter `access_level` on the last change, knowing that this is a breaking change. What you already stated before, maybe this kind of modification is out out scope of a PR.

I will wait with any further modifications until a decision was made, how to handle the problem with the missing parameter `access_level`.

---

In order to revert this change in behavior, it "just" has to be decided which default value to use for `access_level`. The problem here is, that whatever the decision is, we always have a change in behavior. People you used that module only for initial runner creation, got an unprotected runner, whereas people who executed this module more than once got a protected runner. It is a dilemma.

---

See https://github.com/ansible-collections/community.general/pull/5908#discussion_r1089756143, this works around that dilemma (by default there is no change in behavior, but you get a warning and can easily switch to the new behavior already).